src/tools/rest-backend/README.md

```diff
@@ -0,0 +1,191 @@
+Each entry consists of multiple keys (+ meta keys) belonging together,
+while the root key matching a certain schema (regex) is used to identify an entry.
+
+The REST service operates on user-defined repositories (paths).
+This paths are not meant to be used in other tools or applications,
+as the service caches all data in-memory as well.
+Changes to the repositories can therefore result in unexpected and undefined behavior,
+both for the service as well as the stored data.
+
+The reason why a special repository is used to store and retrieve entries is
+that opening the system to everyone by providing a public interface to the
+whole key database would result in a security leak.
+For a REST API with such an interface, another tool will be published.
+
+## Run and Configure ##
+
+To run the `@tool@` we need to find out where it has been installed to.
+This can be done by running the command `kdb list-tools` which will output a list of
+installed tools and something like `External tools are located in /usr/local/lib/elektra/tool_exec`.
+With this path we can run the service like `cd /usr/local/lib/elektra/tool_exec/ && sh run-@tool@`.
+An alternative is to make use of the `kdb` tool and run `kdb run-@tool@`.
+
+The REST service can also be configured. All configuration is read from Elektras
+key database at start-time of the service. Further details and configuration options
+are listed below.
+
+To stop the service, run `sh stop-@tool@` in the directory where the start script is located
+or `kdb stop-@tool@` from anywhere.
+
+### Configuration ###
+
+The service stores all its configuration below `/sw/elektra/@tool@`,
+which is split into two parts:
+
+- one for CppCMS below `/sw/elektra/@tool@/cppcms` and
+- one for the service itself below `/sw/elektra/@tool@/backend`
+
+#### CppCMS ####
+
+All configuration options for CppCMS are listed on their [Website](http://cppcms.com/wikipp/en/page/cppcms_1x_config).
+The JSON configuration explained on the website can be translated into Elektra keys easily.
+For the values `true` and `false`, the strings `"true"` and `"false"` can be used.
+
+The following example configuration in CppCMS style
+```
+{
+    "service": {
+        "api": "http"
+        "port": 8080,
+        "ip": "0.0.0.0"
+    },
+    "security": {
+        "display_error_messages": true
+    }
+}
+```
+can be realized within Elektra like
+```
+kdb set /sw/elektra/@tool@/cppcms/service/api "http"
+kdb set /sw/elektra/@tool@/cppcms/service/port 8080
+kdb set /sw/elektra/@tool@/cppcms/service/ip "0.0.0.0"
+kdb set /sw/elektra/@tool@/cppcms/security/display_error_messages "true"
+```
+
+Simply set the desired settings as keys in the key database and you are done!
+
+#### Backend ####
+
+The service itself offers quite some configuration options as well.
+In detail, the options (without the base key `/sw/elektra/@tool@/backend`) are:
+
+- `api_specification/raw` (string): A link to the blueprint describing the API.
```

Comment:
Can I put this into a separate file that gets somehow mounted/imported during installation (+ maybe added to the readme through `configure_file`)? I'd like to provide a specification, but to be honest, I haven't found out yet how exactly it is done!